scripts/vendor

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+
+set -eu
+
+TYP=$1
+
+usage() {
+  echo "usage: ./scripts/vendor <update|outdated>"
+  exit 1
+}
+
+if [ -z "$TYP" ]; then
+  usage
+fi
+
+# create dummy go.mod, see comment in vendor.mod
+cat > go.mod <<EOL
+module github.com/docker/cli
+
+go 1.16
+EOL
+
+update() {
+  go mod tidy -modfile=vendor.mod
+  go mod vendor -modfile=vendor.mod
+
+  if [ ! -x "$(command -v modvendor)" ]; then
+    echo "modvendor not found. Install with 'go install github.com/goware/modvendor@v0.5.0'"
+    exit 1
+  fi
+  modvendor -copy="**/*.c **/*.h **/*.proto" -v
```

Comment:
@errordeveloper added [modvendor](https://github.com/goware/modvendor) as suggested.

Reply:
The thing I found is that it breaks on Circle CI because of GOPATH being a list and modvendor doesn't support it right now. I had a go at patching it, but didn't manage to get something working and did a simpler thing instead (see https://github.com/docker/swarmkit/pull/3046/commits/e32ee27cbf4a5973dc236e3fdd0f6311403a92b2).

Reply:
In this repo, vendor validation is done inside a container using the official Golang image so there is only one GOPATH thankfully:

```
#11 0.616 GOLANG_VERSION=1.16.11
#11 0.616 GOPATH=/go
#11 0.616 HOME=/root
#11 0.616 MODVENDOR_VERSION=v0.5.0
#11 0.616 OTEL_EXPORTER_OTLP_TRACES_ENDPOINT=unix:///dev/otel-grpc.sock
#11 0.616 OTEL_EXPORTER_OTLP_TRACES_PROTOCOL=grpc
#11 0.616 OTEL_TRACES_EXPORTER=otlp
#11 0.616 OTEL_TRACE_PARENT=00-311c2971a70bf0c057edb57c61a31a22-d9acf1a125c98700-01
#11 0.616 PATH=/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
#11 0.616 PWD=/src
#11 0.616 SHLVL=1
```

Reply:
@thaJeztah did raise a point earlier that it's not clear what license modvendor uses... but perhaps it doesn't matter so much as its a build-only dependency?